src/main/java/com/powsybl/openloadflow/ac/DistributedLoadSlackOuterLoop.java

```diff
@@ -0,0 +1,149 @@
+/**
+ * Copyright (c) 2019, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.openloadflow.ac;
+
+import com.powsybl.commons.PowsyblException;
+import com.powsybl.openloadflow.ac.nr.DefaultNewtonRaphsonStoppingCriteria;
+import com.powsybl.openloadflow.ac.outerloop.OuterLoop;
+import com.powsybl.openloadflow.ac.outerloop.OuterLoopContext;
+import com.powsybl.openloadflow.ac.outerloop.OuterLoopStatus;
+import com.powsybl.openloadflow.network.LfBus;
+import com.powsybl.openloadflow.network.LfNetwork;
+import com.powsybl.openloadflow.network.PerUnit;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Iterator;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * @author Anne Tilloy <anne.tilloy at rte-france.com>
+ */
+public class DistributedLoadSlackOuterLoop implements OuterLoop {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DistributedLoadSlackOuterLoop.class);
+
+    /**
+     * Slack active power residue epsilon: 10^-5 in p.u => 10^-3 in Mw
+     */
+    private static final double SLACK_P_RESIDUE_EPS = Math.pow(10, -5);
+
+    static class ParticipatingLoad {
+
+        final LfBus load;
+
+        double factor;
+
+        ParticipatingLoad(LfBus load, double factor) {
+            this.load = load;
+            this.factor = factor;
+        }
+    }
+
+    @Override
+    public String getName() {
+        return "Distributed slack";
+    }
+
+    private static List<ParticipatingLoad> getParticipatingLoads(LfNetwork network) {
+        return network.getBuses()
+                .stream()
+                .filter(bus -> bus.getLoadTargetP() > 0 && bus.isParticipating())
+                .map(bus -> new ParticipatingLoad(bus, bus.getParticipationFactor()))
```

Comment:
You should use directly getLoadTargetP instead of  getParticipationFactor

Reply:
Done.